src/mscorlib/src/System/IO/Stream.cs

```diff
@@ -767,17 +767,16 @@ public virtual int Read(Span<byte> destination)
 
         // Reads one byte from the stream by calling Read(byte[], int, int). 
         // Will return an unsigned byte cast to an int or -1 on end of stream.
-        // This implementation does not perform well because it allocates a new
-        // byte[] each time you call it, and should be overridden by any 
+        // This implementation should be overridden by any 
         // subclass that maintains an internal buffer.  Then, it can help perf
         // significantly for people who are reading one byte at a time.
         public virtual int ReadByte()
         {
-            byte[] oneByteArray = new byte[1];
-            int r = Read(oneByteArray, 0, 1);
+            Span<byte> bytes = stackalloc byte[1];
```

Comment:
This is, intentionally, a single byte Span but is named as `bytes`. `oneByteSpan` instead?